src/components/GlobalSearch/GlobalSearchComponent.tsx

```diff
@@ -0,0 +1,262 @@
+import React, { useContext, useCallback, useState, useEffect } from 'react';
+import { ClientSDKContext } from '../../context/clientSDKContext';
+import { globalSearch, SearchResource, SearchForFilter } from './globalSearch';
+import { Search } from '../common/Search/Search';
+import { PureObject, ApiQuery } from '../..';
+import _ from 'lodash';
+import styled from 'styled-components';
+import { Icon } from 'antd';
+import { withDefaultTheme } from '../../hoc';
+import { SearchResultList } from './SearchResultList';
+
+class RefCounter {
+  id: number;
+  constructor() {
+    this.id = 0;
+  }
+
+  currentId(): number {
+    return this.id;
+  }
+
+  nextId(): number {
+    return ++this.id;
+  }
+}
+
+export type GlobalSearchStrings = {
+  assets: string;
+  timeseries: string;
+  events: string;
+  files: string;
+  noDescription: string;
+  emptySearch: string;
+  searchPlaceholder: string;
+};
+
+const defaultStrings: GlobalSearchStrings = {
+  assets: 'Assets',
+  timeseries: 'Time Series',
+  events: 'Events',
+  files: 'Files',
+  noDescription: '[No Description]',
+  emptySearch: 'Nothing found',
+  searchPlaceholder: 'Search for a resource',
+};
+
+export type RenderSearchItem = (
+  strings: GlobalSearchStrings,
+  item: SearchResource
+) => React.ReactNode;
+export type ItemSelectedCallback = (item: SearchResource) => void;
+
+type GlobalSearchProps = {
```

Comment:
need to add tsdoc for each property, or at least for some. it's empty in storybook now

Reply:
That didn't help it seems. Storybook doesn't seem to expand nested structures, or am I wrong?

Reply:
nested structures – no, but describe for what some property stands for would be useful (like you did for `searchForFilter`). For instance, it is not clear what's the difference between `renderSearchResult` and `renderSearchItem` from the first look.